### PR TITLE
migrate to iotwebconf 3.0.0

### LINKED ISSUE #357: IotWebConf >= 3.0.0
IotWebConf 3.0.0 has incompatible changes breaking our build:

https://travis-ci.org/github/ecocurious2/MultiGeiger/jobs/751911779

### PR DESCRIPTION
**Pull request by rexfue**
Migration to iotwebconf 3.0.0
Since iotwebconf 3.0.0 is not yet released, we need to directly read the Git repository (in platformio.ini). For the yes/no decisions I use checkboxes now.  Fixes #357

### CODE REVIEW

File: multigeiger/webconf.cpp [hunk 1]
```diff
@@ -17,38 +20,36 @@ bool sendToMadavi = SEND2MADAVI;
...
-IotWebConfParameter deveuiParam = IotWebConfParameter("DEVEUI", "deveui", deveui, 17);
-IotWebConfParameter appeuiParam = IotWebConfParameter("APPEUI", "appeui", appeui, 17);
-IotWebConfParameter appkeyParam = IotWebConfParameter("APPKEY", "appkey", appkey, 33);
+iotwebconf::ParameterGroup grpLoRa = iotwebconf::ParameterGroup("lora", "LoRa Settings");
+iotwebconf::CheckboxParameter sendToLoraParam = iotwebconf::CheckboxParameter("Send to LoRa (=>TTN)", "send2lora", sendToLora_c, CHECKBOX_LEN, sendToLora);
+iotwebconf::TextParameter deveuiParam = iotwebconf::TextParameter("DEVEUI", "deveui", deveui, 17);
+iotwebconf::TextParameter appeuiParam = iotwebconf::TextParameter("APPEUI", "appeui", appeui, 17);
+iotwebconf::TextParameter appkeyParam = iotwebconf::TextParameter("APPKEY", "appkey", appkey, 33);
 
 bool parse_bool(char *text, bool *value) {
```
Comment (ThomasWaldmann) [inline multigeiger/webconf.cpp:54]: is parse_bool still used?

File: multigeiger/webconf.cpp [hunk 2]
```diff
@@ -162,47 +159,39 @@ void configSaved(void) {
...
   isLoraBoard = loraHardware;
   iotWebConf.setConfigSavedCallback(&configSaved);
-  iotWebConf.setupUpdateServer(&httpUpdater);
+  iotWebConf.setupUpdateServer(
+  [](const char *updatePath) {
+    httpUpdater.setup(&server, updatePath);
+  },
+  [](const char *userName, char *password) {
+    httpUpdater.updateCredentials(userName, password);
+  });
```
Comment (ThomasWaldmann) [inline multigeiger/webconf.cpp:171]: this looks a bit strangely formatted.

also, what does `[](...)` even mean? :-)

File: platformio-example.ini [hunk 3]
```diff
@@ -28,5 +28,5 @@ lib_deps=
   Adafruit BME680 Library
   Adafruit BME280 Library
   Adafruit Unified Sensor
-  IotWebConf@<3.0.0
+  https://github.com/prampec/IotWebConf.git
   MCCI LoRaWAN LMIC library
```
Comment (ThomasWaldmann) [inline platformio-example.ini:32]: so this works for platformio, but how about arduino ide users?

File: multigeiger/webconf.cpp (hunk 1 above)
Comment (rexfue) [reply multigeiger/webconf.cpp:54]: No, I will  delete it.

File: multigeiger/webconf.cpp (hunk 2 above)
Comment (rexfue) [reply multigeiger/webconf.cpp:171]: The strange formatting comes from 'astyle'. The code comes from the example or from the migration guide.

File: platformio-example.ini (hunk 3 above)
Comment (rexfue) [reply platformio-example.ini:32]: Arduio users should  load the ZIP from Github and install the zip using library manager.
All this is necessary only as long as there is no official release!

File: multigeiger/webconf.cpp (hunk 2 above)
Comment (ThomasWaldmann) [reply multigeiger/webconf.cpp]: so astyle would complain if you indented the lines?

File: platformio-example.ini (hunk 3 above)
Comment (ThomasWaldmann) [reply platformio-example.ini:32]: the "release" situation is a mess.

otoh, author says it is not officially released, otoh, platformio automatically pulls the 3.0 version already (if not held back by adding `<3.0`).

that arduino users have to manually download and install a zip is also not very convenient.

so I guess we should wait with merging this until everybody gets the new officially really-really-released version.

File: multigeiger/webconf.cpp (hunk 2 above)
Comment (ThomasWaldmann) [reply multigeiger/webconf.cpp]: ah, i see you have solved this already, thanks.

File: platformio-example.ini (hunk 3 above)
Comment (rexfue) [reply platformio-example.ini:32]: I totally agree. Let's wait.

Comment (ThomasWaldmann) [reply platformio-example.ini:32]: https://github.com/prampec/IotWebConf link to check whether 3.0 is officially released.

Comment (ThomasWaldmann) [reply platformio-example.ini:32]: v3.0.1 is now really really officially released! \o/

but @rexfue told at last meeting that this PR needs some more changes to work ok with 3.0.1, so let's wait until these changes are done.

**Comment by rexfue** [comment]
Closed - too old.

**Pull request closed without merging**